### ISSUE TITLE
SVG parser - linear gradient incorrect rendering

### ISSUE THREAD

**Issue by artemp** [mapnik/mapnik#2973]
``` svg
<?xml version="1.0"?>
<svg xmlns="http://www.w3.org/2000/svg" version="1.2" baseProfile="tiny"
     width="800" height="400" viewBox="0 0 800 400" xmlns:xlink="http://www.w3.org/1999/xlink">
  <g>
    <defs>

      <linearGradient id="Base">
        <stop offset="0.05" stop-color="red"/>
        <stop offset="0.95" stop-color="blue"/>
      </linearGradient>
      <linearGradient id="MyGradient" xlink:href="#Base"/>
    </defs>
    <rect fill="url(#MyGradient)" stroke="black" stroke-width="1"
          x="100" y="100" width="600" height="200"/>
  </g>
</svg>
```

expected

<img width="300" alt="safari" src="https://cloud.githubusercontent.com/assets/59056/8680593/aa2af666-2a62-11e5-9bf0-fede76139373.png">

current

![mapnik](https://cloud.githubusercontent.com/assets/59056/8680605/bf267b1c-2a62-11e5-8053-d9a62be193c5.png)

**Comment by artemp** [comment]
Fixed in 0a15a955a2e982b5d789a2a93d864dc8fa8e61c3

**Comment by artemp** [comment]
![linear-gradient](https://user-images.githubusercontent.com/59056/32445528-4e6b9968-c306-11e7-8c54-3f0bd48b5af5.png)
